src/spikeinterface/postprocessing/unit_localization.py

```diff
@@ -451,23 +457,27 @@ def compute_grid_convolution(
         dot_products = np.zeros((weights.shape[0], num_templates), dtype=np.float32)
         for count in range(weights.shape[0]):
             w = weights[count, :, :][channel_mask, :][:, nearest_templates]
-            # w = w / np.sum(w, axis=0)[np.newaxis, None]
-            # w[np.isnan(w)] = 0.
             dot_products[count, :] = np.dot(global_products, w)
 
         dot_products = np.maximum(0, dot_products)
         if percentile < 100:
             thresholds = np.percentile(dot_products, percentile, axis=0)
             dot_products[dot_products < thresholds[np.newaxis, :]] = 0
 
-        found_positions = np.zeros(2, dtype=np.float32)
+        found_positions = np.zeros(ndim, dtype=np.float32)
         scalar_products = np.zeros(num_templates, dtype=np.float32)
         for count in range(weights.shape[0]):
             scalar_products += dot_products[count]
-            found_positions += np.dot(dot_products[count], template_positions[nearest_templates])
+            found_positions[:2] += np.dot(dot_products[count], template_positions[nearest_templates])
 
         unit_location[i, :] = found_positions / scalar_products.sum()
 
+        if mode == "3d":
+            best_template = np.argmin(np.linalg.norm(template_positions - unit_location[i, :2], axis=1))
+            w = weights[:, :, best_template]
+            dot_products = np.dot(w[:, channel_mask], global_products) / np.sum(w, axis=1)
+            unit_location[i, 2] = (dot_products * sigma_um).sum() / dot_products.sum()
```

Comment:
Why is the implementation different to the one in `src/spikeinterface/sortingcomponents/peak_localization.py`?

Reply:
This is the same, modulo the loop over spikes in sortingcomponents